Fix session verification

TobiasFella opened the pull request:
In cases where *we* send start, we never updated the common mac codes, causing the wrong algorithm to be selected

KitsuneRal reviewed (changes requested):
Do we actually need gtad updates here??

TobiasFella commented:
Ah, no, i accidentally commited those

TobiasFella committed:
Fix session verification

In cases where *we* send start, we never updated the common mac codes, causing the wrong algorithm to be selected
